### PR TITLE
GH-127154: Remove PGO from JIT CI

### LINKED ISSUE #127154: Turn off PGO for JIT CI
I don't think we need PGO turned on in JIT CI for non-debug jobs. It's mostly just a waste of time, and doesn't really give us any additional information.

@mdboom, I know that the Windows PGO builds were a nice canary for "breaking" ceval.c under MSVC (since it's not tested in CI or any buildbot), but I figure our weekly benchmarking runs are a good enough "buildbot" for that, so we don't need to run it all the time on PRs.

@savannahostrowski, want to take this?

### Linked PRs
* gh-127212

### PR DESCRIPTION
**Pull request by savannahostrowski**
@brandtbucher I'm assuming you still wanted to keep `--with-lto` wherever that's used but let me know if you'd also like to strip that out.

* Issue: gh-127154

### CODE REVIEW

**Comment by alonme** [comment]
Not totally related,
However it seems to me that [these 2 lines](https://github.com/python/cpython/pull/127212/files#diff-9935012d81892883011d8a947f12458e84471297bb2b7b0e8eab6f6d9e48090bR159-R160) are also unnecessary?

**Comment by savannahostrowski** [comment]
@alonme Hmmm, I don't think so because those lines refer to non-debug builds.

I realized the title of this PR could be misleading, so I updated it to reflect that it is specifically for debug builds.

**Comment by savannahostrowski** [comment]
Haha,  I flipped this...the double negative hurt my brain. Thanks!

**Comment by brandtbucher** [comment]
Nice, let's also drop LTO though (like PGO, it doesn't actually give us any useful new coverage and just takes forever).

How do you feel about also removing the four `*-unknown-linux-gnu/clang` builds while we're at it? Clang is supposed to be ABI-compatible with GCC, so it sort of just seems like we're wasting more time with those builds (and getting Clang to work properly on those builds requires setting the `CC` env var and another tweak or two that I forget).

File: .github/workflows/jit.yml [hunk 1]
```diff
@@ -111,7 +101,7 @@ jobs:
         if: runner.os == 'Windows' && matrix.architecture != 'ARM64'
         run: |
           choco install llvm --allow-downgrade --no-progress --version ${{ matrix.llvm }}.1.0
-          ./PCbuild/build.bat --experimental-jit ${{ matrix.debug && '-d' || '--pgo' }} -p ${{ matrix.architecture }}
+          ./PCbuild/build.bat --experimental-jit ${{ matrix.debug && '-d' || '' }} -p ${{ matrix.architecture }}
           ./PCbuild/rt.bat ${{ matrix.debug && '-d' || '' }} -p ${{ matrix.architecture }} -q --multiprocess 0 --timeout 4500 --verbose2 --verbose3
 
       # No PGO or tests (yet):
```
Comment (brandtbucher) [inline .github/workflows/jit.yml]:
```suggestion
      # No tests (yet):
```

File: .github/workflows/jit.yml [hunk 2]
```diff
@@ -87,18 +85,10 @@ jobs:
...
           - target: aarch64-unknown-linux-gnu/gcc
             architecture: aarch64
             runner: ubuntu-22.04
             compiler: gcc
-          - target: aarch64-unknown-linux-gnu/clang
-            architecture: aarch64
-            runner: ubuntu-22.04
-            compiler: clang
     env:
       CC: ${{ matrix.compiler }}
```
Comment (brandtbucher) [inline .github/workflows/jit.yml]: I don't think we need this anymore (in fact, do we need `compiler` in the matrix above at all?):

```suggestion
```

**Review by brandtbucher — APPROVED** [review]
Nice, this is like a ~40% reduction in resources spent running JIT CI!

Two more suggestions for cleanup:

**Pull request merged**